From 9685193083943df21b09c8edc1f968916b972106 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Fri, 30 Nov 2018 15:29:09 +0200
Subject: [PATCH 0821/1051] net: mvpp2: recycle check ipsec path and dst in skb

Do not recycle skb buffer having the ip-sec-path and refdst fields

Change-Id: I35c8ac0222ca52c2692940d340d08a6179afa4e6
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61441
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 6e46a23e4bb9..81ea93e22ba5 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -3267,6 +3267,9 @@ static int mvpp2_recycle_get_bm_id(struct sk_buff *skb)
 	/* Use skb->cloned but not skb_cloned(), skb_header_cloned() */
 	if (skb_shared(skb) || skb->cloned)
 		return -1;
+	/* ipsec: sp/secpath, _skb_refdst ... */
+	if (!skb_irq_freeable(skb))
+		return -1;
 	if (skb_shinfo(skb)->tx_flags & SKBTX_DEV_ZEROCOPY)
 		return -1;
 
-- 
2.17.1

